src/server/stream_family.cc

```diff
@@ -2297,6 +2309,18 @@ void XReadImpl(CmdArgList args, std::optional<ReadOpts> opts, ConnectionContext*
       if (streamCompareID(&last_id, &requested_sitem.id.val) >= 0) {
         block = false;
       }
+    } else {
+      if (opts->read_group) {
+        // if the key is not found,
+        // return NoGroupOrKey error.
+        // We are simply mimicking Redis' error message here...
+        // However, we could actually report more precise error message...
+        string error_msg =
+            NoGroupOrKey(stream, opts->group_name, " in XREADGROUP with GROUP option");
+        cntx->transaction->Conclude();
```

Comment:
Making a general comment here (no need for action on your side on this PR). 

I really do not like that we conclude transactions manually, it should be covered in some RAII class such that OnScopeExit it concludes itself. Maybe a good solution for this pattern is the Abseil variant called `Cleanup`. And yes I know we conclude before we Send and error but honestly, it won't affect the performance if we literally conclude after we report the error...

Reply:
> I really do not like that we conclude transactions manually, it should be covered in some RAII class such that OnScopeExit it concludes itself.

I was asking @romange about doing RAII for this just a moment... I found out I need to conclude transaction so that unit test won't crash..